sdk/core/azure-core/src/main/java/com/azure/core/util/polling/PollerFlux.java

```diff
@@ -139,6 +155,80 @@ public PollerFlux(Duration pollInterval,
             true);
     }
 
+    /**
+     * Creates PollerFlux.
+     *
+     * This create method uses a {@link PollingStrategy} to poll the status of a long running operation after the
+     * activation operation is invoked. See {@link PollingStrategy} for more details of known polling strategies
+     * and how to create a custom strategy.
+     *
+     * @param pollInterval the polling interval
+     * @param activationOperation the activation operation to activate (start) the long running operation.
+     *     This operation will be invoked at most once across all subscriptions. This parameter is required.
+     *     If there is no specific activation work to be done then invocation should return Mono.empty(),
+     *     this operation will be called with a new {@link PollingContext}.
+     * @param strategy a known strategy for polling a long running operation in Azure
+     * @param resultType the {@link Type} of the final result object to deserialize into, or BinaryData if raw response
+     *                   body should be kept. This should match the generic parameter {@link U}.
+     * @param httpPipeline an instance of {@link HttpPipeline} to send requests with
+     * @param context additional metadata to pass along with the request
+     * @param <U> The type of the final result of long running operation
+     * @return PollerFlux
+     */
+    public static <U> PollerFlux<BinaryData, U>
+        create(Duration pollInterval,
+               Supplier<Mono<? extends Response<?>>> activationOperation,
+               PollingStrategy strategy,
+               Type resultType,
+               HttpPipeline httpPipeline,
+               Context context) {
+        return create(
+            pollInterval,
+            ctx -> activationOperation.get()
+                .flatMap(r -> {
+                    if (!strategy.canPoll(r)) {
+                        return Mono.error(new IllegalStateException("Cannot poll with strategy " + strategy));
```

Comment:
nit, the strategy need to implement `toString` for this error to be useful.

Reply:
A bigger question is how exception should be handled in LLC.

For normal request, no exception as long as there is a response.

For pagination, currently I think it ignores if data model is not a correct pagination.

For LRO, there seems no apparent approach to ignore the error when we cannot find a way to do polling.
However the exception here should be somehow more generic or customizable, as it couples with the handling of this exception outside of `PollerFlux`.

E.g. ARM need to convert the error in polling `ManagementException`.
https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/resourcemanager/azure-resourcemanager-resources/src/main/java/com/azure/resourcemanager/resources/fluentcore/AzureServiceClient.java#L143-L237

Reply:
Also, mgmt uses `PollResult<>` mostly for error handling. We try to get most from the last failed poll result (status code, headers, response body) to give a better `HttpResponseException` to user (for the case when user does not care about the LRO and want to use the API like other non-LRO API).
https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/resourcemanager/azure-resourcemanager-resources/src/main/java/com/azure/resourcemanager/resources/fluentcore/AzureServiceClient.java#L156-L185

Reply:
> For normal request, no exception as long as there is a response.

Actually we still throw (by default). The exception handling in SwaggerMethodParser will default to 2xx for success and 4xx for throw. We do have a way to customize it in `RequestOptions`.

> However the exception here should be somehow more generic or customizable, as it couples with the handling of this exception outside of PollerFlux.

You can do that by throwing your own exception from your `canPoll()` implementation with more information. The exception here is like a fallback exception, if the `PollingStrategy` simply returns `false` without giving more details.

However, I will print out the polling strategy's name instead of the hashcode in the error message.

> Also, mgmt uses PollResult<> mostly for error handling.

In the latest implementation you can pass a polling type `T` into this `PollerFlux.create()`.